File: backend/components/metiers/administration_publique/utils/audit/helpers/audit_helper.test.py
# audit_helper.test.py
# Tests unitaires Python pour audit_helper
from .audit_helper import generate_audit_log

def test_generate_audit_log():
    log = generate_audit_log('LOGIN', {'user': 'alice'})
    assert 'timestamp' in log
    assert log['action'] == 'LOGIN'
    assert log['user'] == 'alice'

File: backend/components/metiers/administration_publique/utils/audit/helpers/audit_helper.test.js
// audit_helper.test.js
// Tests unitaires JS pour audit_helper
const { generateAuditLog } = require('./audit_helper');

describe('generateAuditLog', () => {
  it('génère un log d\'audit structuré', () => {
    const log = generateAuditLog('LOGIN', { user: 'alice' });
    expect(log).toHaveProperty('timestamp');
    expect(log.action).toBe('LOGIN');
    expect(log.user).toBe('alice');
  });
});
